docs/api/kernelspec.rst

```diff
@@ -30,6 +30,10 @@ kernelspec - discovering kernels
       Additional kernel-specific metadata; clients can use this as needed,
       for instance to aid in kernel selection and filtering.
 
+      As with all other metadata fields, jupyter clients should use
+      app-specific submessages as opposed to storing their own top-level
```

Comment:
Add an example, in the hopes that would clarify things. (Happy to tweak more if it doesn't.)